pinpoint/charts/pinpoint-hbase/templates/configmap.yaml

```diff
@@ -0,0 +1,96 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "hbase.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "hbase.name" . }}
+    {{- include "hbase.labels" . | nindent 4 }}
+data:
+  bootstrap.sh: |
+    #!/bin/bash
+
+    : ${HBASE_PREFIX:=/opt/hbase/hbase-1.2.6}
+
+    . $HBASE_PREFIX/conf/hbase-env.sh
+
+    # Directory to find config artifacts
+    CONFIG_DIR="/tmp/hbase-config"
+
+    # Copy config files from volume mount
+
+    for f in hbase-site.xml hbase-env.sh; do
+      if [[ -e ${CONFIG_DIR}/$f ]]; then
+        cp ${CONFIG_DIR}/$f $HBASE_PREFIX/conf/$f
+      else
+        echo "ERROR: Could not find $f in $CONFIG_DIR"
+        exit 1
+      fi
+    done
+
+    if [[ $2 == "master" ]]; then
+      $HBASE_PREFIX/bin/hbase-daemon.sh start master
+      $HBASE_PREFIX/bin/hbase-daemon.sh start thrift
+    fi
+
+    if [[ $2 == "regionserver" ]]; then
+      #  wait up to 30 seconds for masternode
+      (while [[ $count -lt 15 && -z `curl -sf http://{{ include "hbase.fullname" . }}-master:16010` ]]; do ((count=count+1)) ; echo "Waiting for {{ include "hbase.fullname" . }}-master" ; sleep 2; done && [[ $count -lt 15 ]])
+      [[ $? -ne 0 ]] && echo "Timeout waiting for hbase-master, exiting." && exit 1
+      $HBASE_PREFIX/bin/hbase-daemon.sh start regionserver
+    fi
+    
+    if echo -e "exists 'HostApplicationMap_Ver2'" | ${HBASE_PREFIX}/bin/hbase shell 2>&1 | grep -q "does exist" 2>/dev/null
+    then 
+      echo "Tables already exist"
+    else
+      sleep 15        
+      echo "create tables"
+      chmod 777 /opt/hbase/hbase-create.hbase
+      ${HBASE_PREFIX}/bin/hbase shell /opt/hbase/hbase-create.hbase
+    fi
+
+    if [[ $1 == "-d" ]]; then
+      until find ${HBASE_PREFIX}/logs -mmin -1 | egrep -q '.*'; echo "`date`: Waiting for logs..." ; do sleep 2 ; done
+      tail -F ${HBASE_PREFIX}/logs/* &
+      while true; do sleep 1000; done
+    fi
+
+    if [[ $1 == "-bash" ]]; then
+      /bin/bash
+    fi
+  hbase-env.sh: |
+    # Extra Java runtime options.
+    # Below are what we set by default.  May only work with SUN JVM.
+    # For more on why as well as other possible settings,
+    # see http://hbase.apache.org/book.html#performance
+    export HBASE_OPTS="$HBASE_OPTS -XX:+UseConcMarkSweepGC"
+  hbase-site.xml: |
```

Comment:
Is this going to override hbase-site.xml inside the dockerfile?

Reply:
No, It doesn't be overridden. I thought that **hbase-site.xml** in dockerfile isn't appropriate for hbase in kubernetes. Because, **hbase-site.xml** in dockerfile is for docker-compose. 
However, if you(or user) want to change the **hbase-site.xml**, you can change the values except **hbase.master**, **hbase.cluster.distributed** using **hbaseSite field in values.yaml.**

Reply:
You are absolutenly correct. **hbase-site.xml** in dockerfile isn't appropriate for hbase in kubernetes. It's standalone with peuso-distributed(if I'm remembering it correctly). So it needs to be overridden with the one in configmap.yaml. 
Or.... maybe I'm misunderstanding something?

Reply:
I think I said it wrong. The file which is finally running is the hbase-site.xml in the helm chart.

In more detail, It's not overriding. Because pinpoint-hbase image has hbase-site.xml in ${HBASE_HOME}/conf/ and configMap in helm chart bind to /tmp/hbase-site.xml. But helm chart watches the /tmp/hbase-site.xml, the /tmp/hbase-site.xml is finally applied.)

Reply:
Great, then I think it makes sense. Just wondering which part of code makes the hbase in helm chart to point to  _/tmp/hbase-site.xml_ ?